### ISSUE TITLE
PVC jsonpath={.status.capacity.storage} isn't update after expand (resize) pvc.

### ISSUE THREAD

**Issue by napat1412** [kubernetes/kubernetes#65044]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>

/kind bug
> /kind feature

/sig storage

**What happened**:
pvc jsonpath={.status.capacity.storage} isn't update after expand (resize) pvc.

**What you expected to happen**:
After pvc is expended, pvc jsonpath={.status.capacity.storage} must equal to jsonpath={.spec.resources.requests.storage}.

**How to reproduce it (as minimally and precisely as possible)**:
I create storageclass to use rbd volume. This is ceph-sc.yaml
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: default
provisioner: kubernetes.io/rbd
parameters:
  monitors: 172.16.0.181:6789, 172.16.0.182:6789, 172.16.0.183:6789
  adminId: admin
  adminSecretName: ceph-secret
  adminSecretNamespace: default
  pool: rbd
  userId: admin
  userSecretName: ceph-secret
  userSecretNamespace: default
  fsType: ext4
  imageFormat: "2"
  imageFeatures: layering
allowVolumeExpansion: true
```

I create pvc myclaim.yaml (pvc size 3Gb).
```
$ kubectl get pv,pvc
NAME                                                        CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS    CLAIM             STORAGECLASS   REASON    AGE
persistentvolume/pvc-f8d80182-6ebb-11e8-805c-000c29fa6222   3Gi        RWO            Delete           Bound     default/myclaim   default                  3s
NAME                            STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
persistentvolumeclaim/myclaim   Bound     pvc-f8d80182-6ebb-11e8-805c-000c29fa6222   3Gi        RWO            default        10s
```

$ kubectl get pvc myclaim -oyaml
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/rbd
  creationTimestamp: 2018-06-13T03:43:44Z
  finalizers:
  - kubernetes.io/pvc-protection
  name: myclaim
  namespace: default
  resourceVersion: "177461"
  selfLink: /api/v1/namespaces/default/persistentvolumeclaims/myclaim
  uid: f8d80182-6ebb-11e8-805c-000c29fa6222
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 3Gi
  storageClassName: default
  volumeName: pvc-f8d80182-6ebb-11e8-805c-000c29fa6222
status:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: 3Gi
  phase: Bound
```

Then, i expand pvc from 3Gi to 5Gi.
$ kubectl edit pvc myclaim
persistentvolumeclaim "myclaim" edited

$ kubectl create -f mycliam-pod.yaml
$ kubectl exec myclaim df
```
Filesystem           1K-blocks      Used Available Use% Mounted on
overlay               47081000  10253992  36827008  22% /
tmpfs                  1940248         0   1940248   0% /dev
tmpfs                  1940248         0   1940248   0% /sys/fs/cgroup
/dev/rbd0              5029504     20472   4730504   0% /mnt
/dev/mapper/cl-root   47081000  10253992  36827008  22% /dev/termination-log
/dev/mapper/cl-root   47081000  10253992  36827008  22% /etc/resolv.conf
/dev/mapper/cl-root   47081000  10253992  36827008  22% /etc/hostname
/dev/mapper/cl-root   47081000  10253992  36827008  22% /etc/hosts
shm                      65536         0     65536   0% /dev/shm
/dev/mapper/cl-root   47081000  10253992  36827008  22% /run/secrets
tmpfs                  1940248        12   1940236   0% /var/run/secrets/kubernetes.io/serviceaccount
tmpfs                  1940248         0   1940248   0% /proc/kcore
tmpfs                  1940248         0   1940248   0% /proc/timer_list
tmpfs                  1940248         0   1940248   0% /proc/timer_stats
tmpfs                  1940248         0   1940248   0% /proc/sched_debug
tmpfs                  1940248         0   1940248   0% /proc/scsi
tmpfs                  1940248         0   1940248   0% /sys/firmware
```
At this point, my storage in pvc is expanded to new size (5Gi)

$ kubectl get pv,pvc
```
NAME                                                        CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS    CLAIM             STORAGECLASS   REASON    AGE
persistentvolume/pvc-f8d80182-6ebb-11e8-805c-000c29fa6222   5Gi        RWO            Delete           Bound     default/myclaim   default                  49m

NAME                            STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
persistentvolumeclaim/myclaim   Bound     pvc-f8d80182-6ebb-11e8-805c-000c29fa6222   3Gi        RWO            default        50m
```
The pv is expand to new size (5Gi) but pvc still is 3Gi.

$ kubectl get pvc myclaim -oyaml
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-provisioner: kubernetes.io/rbd
  creationTimestamp: 2018-06-13T03:43:44Z
  finalizers:
  - kubernetes.io/pvc-protection
  name: myclaim
  namespace: default
  resourceVersion: "177660"
  selfLink: /api/v1/namespaces/default/persistentvolumeclaims/myclaim
  uid: f8d80182-6ebb-11e8-805c-000c29fa6222
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
  storageClassName: default
  volumeName: pvc-f8d80182-6ebb-11e8-805c-000c29fa6222
status:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: 3Gi
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2018-06-13T03:46:20Z
    message: Waiting for user to (re-)start a pod to finish file system resize of
      volume on node.
    status: "True"
    type: FileSystemResizePending
  phase: Bound
```
$ kubectl get pvc myclaim --output jsonpath={.spec.resources.requests.storage}
5Gi
$ kubectl get pvc myclaim --output jsonpath={.status.capacity.storage}
3Gi

**I get yaml of pvc myclaim again & I found new jsonpath={.status.conditions}.
I think kubernetes update pvc jsonpath={.status.conditions}. But kubernetes not update pvc jsonpath={.spec.capacity.storage}. Why? or I do some think wrong.**

**Anything else we need to know?**:
I build kube-controller-manager that install ceph-common package.

**Environment**:
- Kubernetes version (use `kubectl version`): kubectl v1.10.4
- Cloud provider or hardware configuration: bare metal (kubeadm)
- OS (e.g. from /etc/os-release): 
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"
CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

- Kernel (e.g. `uname -a`): Linux m1.service.local 3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kubeadm v1.10.4
- Others: ceph luminous 12.2.5 for rbd storageclass

**Comment by napat1412** [comment]
This comment explain how to clean status of Persistent Volume Claim(PVC).
https://github.com/kubernetes/kubernetes/issues/35941#issuecomment-328028093 (In appendixes)
Do i manual clean status of every pvc with etcd?

**Comment by msau42** [comment]
@napat1412 once the PV has been resized, you need to restart your Pod in order to get the filesystem to resize, and PVC size to be updated

**Comment by napat1412** [comment]
@msau42  I delete my pod & start my pod but my PVC size still is not update.
Could you give me the example of kube-controller-manager.yaml and storageclass.yaml.

**Comment by kaveri-s-miq** [comment]
@napat1412 Hey, we had faced the same issue some time back. This issue is resolved in kubernetes 1.11. One keynote would be to make sure the worker node ami version is also upgraded to 1.11. That will most likely solve your issue.
